ui/src/app/views/workspace-library.tsx

```diff
@@ -95,22 +99,41 @@ export const WorkspaceLibrary = withUserProfile()
   updateListedWorkspaces(tabEnum) {
     switch (tabEnum) {
       case tabEnum === libraryTabEnums.PUBLISHED_WORKSPACES:
-        this.reloadPublishedWorkspaces();
+        this.loadPublishedWorkspaces();
         break;
       default:
+        this.loadFeaturedWorkspaces();
         break;
     }
   }
 
-  async reloadPublishedWorkspaces() {
-    this.setState({workspacesLoading: true});
+  async loadFeaturedWorkspaces() {
+    try {
+      const resp = await featuredWorkspacesConfigApi().getFeaturedWorkspacesConfig();
+      const idToNamespace = new Map();
+      resp.featuredWorkspacesList.map(fw => idToNamespace.set(fw.id, fw.namespace));
+
+      this.setState({
+        featuredWorkspaces: this.state.workspaceList.filter(ws => idToNamespace.get(ws.workspace.id)
+          && idToNamespace.get(ws.workspace.id) === ws.workspace.namespace),
+        workspacesLoading: false
```

Comment:
[Spinner nitpick]

I'm getting a little confused on the exact expected behavior of 'workspacesLoading' and display of the spinner.

Just thinking through the loading paths:

1) On component mount, we do [loadPublished --> loadFeatured]

2) When the user clicks refresh, we do [loadPublished].

For (1), I would expect the spinner to show until both calls are complete. But I think the spinner will be hidden once loadPublished is complete, ignoring loadFeatured. Is that correct?

For (2), I would expect some spinner to show while the API call is being made. But right now I think the refresh will happen without user feedback.

One way to possibly handle this would be to have loadPublished setState to {loadingWorkspaces: true} when first called, which would at least make case (2) show a spinner while loading.

Reply:
that's what i was doing with it, but the workspaces load so quickly upon reload that the spinner basically just flashes on the screen.  isn't a great UX, although maybe it is better to have it implemented that way. thoughts?

Reply:
I'm surprised workspaces are loading so quickly on load! In the user-specific workspace page it's been taking a few seconds regularly.

I agree, a super-quick spinner isn't something we should keep. Feel free to close this thread out then, & we can always follow-up with a UX tweak later on if needed.